pkg/api/types.go

```diff
@@ -3312,6 +3314,31 @@ const (
 	TLSCertKey = "tls.crt"
 	// TLSPrivateKeyKey is the key for the private key field in a TLS secret.
 	TLSPrivateKeyKey = "tls.key"
+
+	// SecretBootstrapToken is used during the automated bootstrap process (first
+	// implemented by kubeadm). It stores tokens that are used to sign well known
+	// ConfigMaps. They may also eventually be used for authentication.
+	SecretTypeBootstrapToken SecretType = "bootstrap.kubernetes.io/token"
```

Comment:
I don't think we have clarity as to best practices here.  There has been discussion about the pain of having alpha/beta in annotations.  That is probably the most similar situation here.

Regardless, what is the right way to graduate this stuff?  As we move to GA we could have the controllers look for secrets that are of either the beta or GA type.  That would complicate the controller.  I *think* we can make it work with a set based selector but it isn't clear if/how that is supported on field selectors yet.

  @mikedanese -- do you know the latest on that?